acquire-driver-common/src/simcams/simulated.camera.c

```diff
@@ -376,8 +404,11 @@ simcam_set(struct Camera* camera, struct CameraProperties* settings)
     };
 
     size_t nbytes = aligned_bytes_of_image(shape);
-    self->im.data = malloc(nbytes);
-    EXPECT(self->im.data, "Allocation of %llu bytes failed.", nbytes);
+    self->im.frame_data = malloc(nbytes);
+    EXPECT(self->im.frame_data, "Allocation of %llu bytes failed.", nbytes);
+
+    self->im.render_data = malloc(nbytes);
+    EXPECT(self->im.render_data, "Allocation of %llu bytes failed.", nbytes);
```

Comment:
`simcam_set` can get called multiple times and should be thread safe.

This will leak memory when every time set is called.

There should probably be a lock around the whole function. `simcam_execute_trigger` may need to get factored so we avoid recursive locking. The memory allocation definitely needs to be inside an acquired `self->im.lock`.

Reply:
I think we can get away with locking just after executing the trigger. There aren't any read or write operations on `self` before that time, beyond just getting the pointer.